### PR TITLE
executor: remove Next function for UnionScanExec

### PR DESCRIPTION
**Pull request by yangwenmai**

### CODE REVIEW

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
to #5985

**Comment by zz-jason** [comment]
@yangwenmai plz fix ci

**Comment by yangwenmai** [comment]
I passed all the local tests.
But I got "*** Test killed with quit: ran too long (10m0s)."  in those CI, Can you rebuild once？

**Comment by zz-jason** [comment]
@yangwenmai there is panic:
```txt
PANIC: ddl_test.go:87: testStatsCacheSuite.TestDDLHistogram

... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0x1058CCA)

/Users/jianzhang.zj/opt/goroot/src/runtime/panic.go:505
  in gopanic
/Users/jianzhang.zj/opt/goroot/src/runtime/panic.go:63
  in panicmem
/Users/jianzhang.zj/opt/goroot/src/runtime/signal_unix.go:388
  in sigpanic
ddl.go:108
  in Handle.insertColStats2KV
ddl.go:36
  in Handle.HandleDDLEvent
ddl_test.go:99
  in testStatsCacheSuite.TestDDLHistogram
/Users/jianzhang.zj/opt/goroot/src/reflect/value.go:308
  in Value.Call
/Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/vendor/github.com/pingcap/check/check.go:798
  in suiteRunner.forkTest.func1
/Users/jianzhang.zj/opt/gopath/src/github.com/pingcap/tidb/vendor/github.com/pingcap/check/check.go:692
  in suiteRunner.forkCall.func1
/Users/jianzhang.zj/opt/goroot/src/runtime/asm_amd64.s:2361
  in goexit
```

File: statistics/ddl.go [hunk 1]
```diff
@@ -100,12 +100,12 @@ func (h *Handle) insertColStats2KV(tableID int64, colInfo *model.ColumnInfo) err
...
 		}
-		var row types.Row
-		row, err = rs[0].Next(ctx)
+		chk := rs[0].NewChunk()
+		err = rs[0].NextChunk(ctx, chk)
 		if err != nil {
 			return errors.Trace(err)
 		}
-		count := row.GetInt64(0)
+		count := chk.NumRows()
```
Comment (lamxTyler) [inline statistics/ddl.go]: `NumRows` is the number of rows in this chunk while `count` is the first value in the first row.

**Comment by XuHuaiyu** [comment]
Hi, @yangwenmai please address the comment

File: statistics/ddl.go (hunk 1 above)
Comment (yangwenmai) [reply statistics/ddl.go]: I still haven't found how to convert it, I also need some time to understand.

**Comment by yangwenmai** [comment]
@XuHuaiyu PTAL

**Pull request merged**